File: src/main/java/jukeboxmc/block/BlockSand.java
package jukeboxmc.block;

import org.jukeboxmc.block.type.SandType;
import org.jukeboxmc.item.ItemSand;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockSand extends Block {

    public BlockSand() {
        super( "minecraft:sand" );
    }

    @Override
    public ItemSand toItem() {
        return new ItemSand( this.runtimeId );
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.SAND;
    }

    public BlockSand setSandType( SandType sandType ) {
       return this.setState( "sand_type", sandType.name().toLowerCase() );
    }

    public SandType getSandType() {
        return this.stateExists( "sand_type" ) ? SandType.valueOf( this.getStringState( "sand_type" ) ) : SandType.NORMAL;
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockSand.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.Sand
import org.jukeboxmc.api.block.data.SandType
import org.jukeboxmc.server.block.JukeboxBlock

class BlockSand(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), Sand {

    override fun getSandType(): SandType {
        return SandType.valueOf(this.getStringState("sand_type"))
    }

    override fun setSandType(value: SandType): Sand {
        return this.setState("sand_type", value.name.lowercase())
    }
}